core/src/message/mod.rs

```diff
@@ -137,6 +138,23 @@ impl DispatchKind {
     pub fn is_signal(&self) -> bool {
         matches!(self, Self::Signal)
     }
+
+    /// Sys-calls that are not allowed to be called for the dispatch kind.
+    pub fn forbidden_funcs(&self) -> BTreeSet<&'static str> {
+        match self {
+            DispatchKind::Signal => [
+                "gr_source",
```

Comment:
@SabaunT someone replace this set with SysCallName maybe? Sync on it, please

Reply:
Will be done in another PR as it requires some refactoring